packages/grid/_modules_/grid/hooks/features/rows/useGetRowId.ts

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { ApiRef } from '../../../models/api/apiRef';
+import { RowData, RowIdGetter } from '../../../models/rows';
+import { useApiMethod } from '../../root/useApiMethod';
+
+const getDefaultRowId = (row: RowData) => row.id;
+
+export const useGetRowId = (apiRef: ApiRef, getRowIdProp?: RowIdGetter) => {
+  const getRowId = React.useCallback(
+    (row: RowData) => {
+      if (getRowIdProp != null) {
+        return getRowIdProp(row);
+      }
+      return getDefaultRowId(row);
+    },
+    [getRowIdProp],
+  );
```

Comment:
Can you inverse the branching? We don't need to test each time if the prop is defined or not. We can do the test only once.